### PR TITLE
[Layer] Convert static MMHelper class to instance Class in DecoderContext.

### PR DESCRIPTION
**Pull request by changqi1**

### CODE REVIEW

File: src/searchers/beam_search.cpp [hunk 1]
```diff
@@ -392,7 +396,11 @@ void BeamSearch::searchTopK(std::tuple<float *, int, int> &result) {
         std::vector<long unsigned int> recvCount(msgerSize, static_cast<long unsigned int>(batchSize * numBeams));
         messenger.allgatherv(maxVal, batchSize * numBeams, recvMax, recvCount);
 
-        float sumVal[batchSize * numBeams] = {0};
+        float sumVal[batchSize * numBeams];
+        for (int i = 0; i < batchSize * numBeams; ++i) {
+            sumVal[i] = 0.0f;
```
Comment (pujiang2018) [inline src/searchers/beam_search.cpp:401]: Original float "sumVal[batchSize * numBeams] = {0};" should work.

File: src/searchers/beam_search.cpp [hunk 2]
```diff
@@ -373,7 +373,11 @@ void BeamSearch::searchTopK(std::tuple<float *, int, int> &result) {
     // 4. add beam socre. Initialize -1e9 to all beams except the first one
     if (msgerSize > 1) {
         // Get the maximum value of each beam through all instance
-        float maxVal[batchSize * numBeams] = {std::numeric_limits<float>::lowest()};
+        float maxVal[batchSize * numBeams];
+        for (int i = 0; i < batchSize * numBeams; ++i) {
+            maxVal[i] = std::numeric_limits<float>::lowest();
```
Comment (pujiang2018) [inline src/searchers/beam_search.cpp:378]: I guess the author intended to set all values to lowest float. good fix.

**Review by pujiang2018 — COMMENTED** [review]
Guess the performance and accuracy not impacted, but let's confirm it before merge.

**Review by pujiang2018 — APPROVED** [review]

File: src/searchers/beam_search.cpp (hunk 1 above)
Comment (changqi1) [reply src/searchers/beam_search.cpp:401]: Yes, original code could work, but this is c++20 standard. If we want to build w/ ICX, it should be c++17 standard.

File: src/searchers/beam_search.cpp (hunk 2 above)
Comment (changqi1) [reply src/searchers/beam_search.cpp:378]: Yes, original code could work, but this is c++20 standard. If we want to build w/ ICX, it should be c++17 standard.

**Pull request merged**